lib/matplotlib/patches.py

```diff
@@ -3279,19 +3308,54 @@ def _get_arrow_wedge(self, x0, y0, x1, y1,
 
             return vertices_arrow, codes_arrow, ddx, ddy
 
+        def _get_bracket(self, x0, y0,
+                         cos_t, sin_t, width, length, angle):
+
+            # arrow from x0, y0 to x1, y1
+            from matplotlib.bezier import get_normal_points
+            x1, y1, x2, y2 = get_normal_points(x0, y0, cos_t, sin_t, width)
+
+            dx, dy = length * cos_t, length * sin_t
+
+            vertices_arrow = [(x1 + dx, y1 + dy),
+                              (x1, y1),
+                              (x2, y2),
+                              (x2 + dx, y2 + dy)]
+            codes_arrow = [Path.MOVETO,
+                           Path.LINETO,
+                           Path.LINETO,
+                           Path.LINETO]
+
+            if angle:
+                trans = transforms.Affine2D().rotate_deg_around(x0, y0, angle)
+                vertices_arrow = trans.transform(vertices_arrow)
+
+            return vertices_arrow, codes_arrow
+
         def transmute(self, path, mutation_size, linewidth):
 
-            head_length = self.head_length * mutation_size
-            head_width = self.head_width * mutation_size
-            head_dist = np.hypot(head_length, head_width)
-            cos_t, sin_t = head_length / head_dist, head_width / head_dist
+            if self.beginarrow_curve or self.endarrow_curve:
+                head_length = self.head_length * mutation_size
+                head_width = self.head_width * mutation_size
+                head_dist = np.hypot(head_length, head_width)
+                cos_t, sin_t = head_length / head_dist, head_width / head_dist
+
+            if self.scaleA is None:
+                scaleA = mutation_size
+            else:
+                scaleA = self.scaleA
+
+            if self.scaleB is None:
+                scaleB = mutation_size
+            else:
+                scaleB = self.scaleB
```

Comment:
Should fit, I think.
```suggestion
            scaleA = mutation_size if self.scaleA is None else self.scaleA
            scaleB = mutation_size if self.scaleB is None else self.scaleB
```